benchmarks/bm/_scenario.py

```diff
@@ -0,0 +1,72 @@
+import abc
+import time
+
+import attr
+import pyperf
+import six
+
+
+def register(cls):
+    """Decorator for scenario class that registers the scenario to be run."""
+    wrapped_cls = attr.s(cls)
+    _register(wrapped_cls)
+    return wrapped_cls
+
+
+var = attr.ib
+
+
+@attr.s
+class Scenario(six.with_metaclass(abc.ABCMeta)):
```

Comment:
the way you could make the auto-register part work is to have a:

```python
class ScenarioMeta(type):
    def __init__(cls, name, bases, dict):       
        # TODO: Modify name/bases/dict?
        new_cls = super().__init__(name, bases, dict)
        _register(new_cls)
       return new_cls


class Scenario(six.with_metaclass(ScenarioMeta)):
    pass
```

(ush, untested)

Reply:
Could this metaclass be combined with `abc` to get the benefit of requiring implementation of `run`?

Reply:
I _think_ the way to do it would be:

```python
class ScenarioMeta(abs.ABCMeta):
    def __init__(cls, ...): ...   
```

Since a meta class is an inheritance from a class, which ABCMeta is... ?